R/oc_check_query.R

```diff
@@ -0,0 +1,192 @@
+#'   values between 1 and 100 are allowed.
+#' @inheritParams oc_forward
+#'
+#' @keywords internal
+
+
+oc_check_query <-
+  function(
+    placename = NULL,
+    latitude = NULL,
+    longitude = NULL,
+    key = NULL,
+    bounds = NULL,
+    countrycode = NULL,
+    language = NULL,
+    limit = NULL,
+    min_confidence = NULL,
+    no_annotations = NULL,
+    no_dedupe = NULL,
+    no_record = NULL,
+    abbrv = NULL,
+    add_request = NULL
+  ) {
+    arglist <-
+      purrr::compact(
+        list(
+          placename = placename,
+          latitude = latitude,
+          longitude = longitude,
+          bounds = bounds,
+          countrycode = countrycode,
+          language = language,
+          limit = limit,
+          min_confidence = min_confidence,
+          no_annotations = no_annotations,
+          no_dedupe = no_dedupe,
+          abbrv = abbrv,
+          add_request = add_request
+        )
+      )
+
+    # prevent obscure warning message from pwalk if length(arglist) == 0
+    stopifnot(length(arglist) >= 1)
+
+    purrr::pwalk(
+      .l = arglist,
+      .f = .oc_check_query,
+      key = key,
+      no_record = no_record
+    )
+  }
+
+.oc_check_query <-
+  function(
+    placename = NULL,
+    latitude = NULL,
+    longitude = NULL,
+    key = NULL,
+    bounds = NULL,
+    countrycode = NULL,
+    language = NULL,
+    limit = NULL,
+    min_confidence = NULL,
+    no_annotations = NULL,
+    no_dedupe = NULL,
+    no_record = NULL,
+    abbrv = NULL,
+    add_request = NULL
+  ) {
+    # check placename
+    if (!is.null(placename)) {
+      if (!is.character(placename)) {
+        stop(call. = FALSE, "`placename` must be a character vector.")
+      }
+    }
+
+    # check latitude
+    if (!is.null(latitude)) {
+      if (!(dplyr::between(latitude, -90, 90))) {
+        stop(call. = FALSE, "`latitude` must be between -90 and 90.")
+      }
+    }
+
+    # check longitude
+    if (!is.null(longitude)) {
+      if (!(dplyr::between(longitude, -180, 180))) {
+        stop(call. = FALSE, "`longitude` must be between -180 and 180.")
+      }
+    }
+
+    # check key
+    if (is.null(key)) {
+      stop(call. = FALSE, "A `key` must be provided.")
+    } else if (!is.character(key)) {
+      stop(call. = FALSE, "`key` must be a character vector.")
+    } else if (length(key) > 1) {
+      stop(call. = FALSE, "`key` must be a vector of length one.")
+    }
+
+    # check bounds
+    if (!is.null(bounds)) {
+      if (length(bounds) != 4) {
+        stop(
+          call. = FALSE,
+          "Every `bbox` must be a numeric vector of length 4."
+        )
+      }
+      oc_check_bbox(bounds[[1]], bounds[[2]], bounds[[3]], bounds[[4]])
+    }
+
+    # check countrycode
+    if (!is.null(countrycode)) {
+      if (!(all(toupper(unlist(countrycode)) %in% countrycodes$code))) {
+        stop(call. = FALSE, "`countrycode` does not have a valid value.")
```

Comment:
same here, what should be the phrasing so that it might work for both single and multiple value countrycodes?

Besides, should we check that countrycode is a list/an single character?

Reply:
I made some changes. Is this better?

> should we check that countrycode is a list/an single character

not entirely sure what you mean. multiple countrycodes are allowed, e.g. french territories. Or do you mean, check that it is a (list of) character vector(s)? Do you have an example that is invalid but not picked up as such yet?